replication.proto

```diff
@@ -0,0 +1,131 @@
+  // This field is REQUIRED.
+  // This field MUST contain enough information to uniquely identify
+  // this specific volume vs all other volumes supported by this plugin.
+  // This field SHALL be used by the CO in subsequent calls to refer to
+  // this volume.
+  string volume_id = 1;
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 2;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 3 [(replication_secret) = true];
+}
+
+message EnableVolumeReplicationResponse {
+}
+
+
+message DisableVolumeReplicationRequest {
+  // The identifier for this volume, generated by the plugin during
+  // CreateVolume CSI RPC call.
+  // This field is REQUIRED.
+  // This field MUST contain enough information to uniquely identify
+  // this specific volume vs all other volumes supported by this plugin.
+  // This field SHALL be used by the CO in subsequent calls to refer to
+  // this volume.
+  string volume_id = 1;
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 2;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 3 [(replication_secret) = true];
+}
+
+message DisableVolumeReplicationResponse {
+}
+
+message PromoteVolumeRequest {
+  // The identifier for this volume, generated by the plugin during
+  // CreateVolume CSI RPC call.
+  // This field is REQUIRED.
+  // This field MUST contain enough information to uniquely identify
+  // this specific volume vs all other volumes supported by this plugin.
+  // This field SHALL be used by the CO in subsequent calls to refer to
+  // this volume.
+  string volume_id = 1;
+  // This field is optional.
+  // Default value is false, force option to Promote the volume.
+  bool  force = 2;
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 3;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 4 [(replication_secret) = true];
+}
+
+message PromoteVolumeResponse{
+}
+
+message DemoteVolumeRequest {
+  // The identifier for this volume, generated by the plugin during
+  // CreateVolume CSI RPC call.
+  // This field is REQUIRED.
+  // This field MUST contain enough information to uniquely identify
+  // this specific volume vs all other volumes supported by this plugin.
+  // This field SHALL be used by the CO in subsequent calls to refer to
+  // this volume.
+  string volume_id = 1;
+  // This field is optional.
+  // Default value is false, force option to Demote the volume.
+  bool  force = 2;
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 3;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 4 [(replication_secret) = true];
+}
+
+message DemoteVolumeResponse{
+}
+
+message ResyncVolumeRequest {
```

Comment:
If we are reserving 2 for something else lets mark 2 as reserved

Reply:
I guess you missed this one

Reply:
done now.